--- configure.ac.orig	2019-03-27 07:29:13 +0000
+++ configure.ac	2019-03-27 07:29:39 +0000
@@ -41,11 +41,12 @@
 AC_PROG_LIBTOOL
 
 # Require X.Org macros 1.8 or later for AC_PROG_INSTALL
-m4_ifndef([XORG_MACROS_VERSION],
-          [m4_fatal([must install xorg-macros 1.8 or later before running autoconf/autogen])])
-XORG_MACROS_VERSION(1.8)
-XORG_DEFAULT_OPTIONS
-XORG_CHECK_MALLOC_ZERO
+#m4_ifndef([XORG_MACROS_VERSION],
+#          [m4_fatal([must install xorg-macros 1.8 or later before running autoconf/autogen])])
+#XORG_MACROS_VERSION(1.8)
+#XORG_DEFAULT_OPTIONS
+#XORG_CHECK_MALLOC_ZERO
+AC_PROG_INSTALL
 
 # Check render configuration, strip extra digits from package version to
 # find the required protocol version
