--- onig-6.9.1/regcomp.c.orig	2019-04-24 07:38:23 +0000
+++ onig-6.9.1/regcomp.c	2019-04-24 07:40:17 +0000
@@ -6346,19 +6346,22 @@
          OnigErrorInfo* einfo)
 {
   int r;
+  regex_t *regg;
 
-  *reg = (regex_t* )xmalloc(sizeof(regex_t));
-  if (IS_NULL(*reg)) return ONIGERR_MEMORY;
+  regg = (regex_t* )xmalloc(sizeof(regex_t));
+  if (IS_NULL(regg)) return ONIGERR_MEMORY;
 
-  r = onig_reg_init(*reg, option, ONIGENC_CASE_FOLD_DEFAULT, enc, syntax);
+  r = onig_reg_init(regg, option, ONIGENC_CASE_FOLD_DEFAULT, enc, syntax);
   if (r != 0) goto err;
 
-  r = onig_compile(*reg, pattern, pattern_end, einfo);
+  r = onig_compile(regg, pattern, pattern_end, einfo); 
   if (r != 0) {
   err:
-    onig_free(*reg);
-    *reg = NULL;
+    onig_free(regg);
+    regg = NULL;
   }
+
+  *reg = regg;
   return r;
 }
 
